### DISCUSSION TITLE
Is it possible to do partial diffusion on a single loop in an interface?

### DISCUSSION THREAD

**Discussion by ostannick** [RosettaCommons/RFdiffusion#184] [Q&A]
If I have a protein complex A:B, and I want to diffuse non-contiguous segments of B (at the positions where it forms an interface with A, i.e. loops), what is the hydra syntax for this?

The partial diffusion examples mention diffusion for the end of a chain, or an entire chain, but not an internal segment of a chain. Can RFDiffusion do this?

Thank you in advance.

**Comment by roccomoretti** [comment]
RFdiffusion works with residues on a "residue gas" basis. Chemical connectivity is imposed as a conditional bias on the inter-residue separation. (That is, if you have two polymerically connected residues, the probability of the location C-term residue is strongly peaked "downstream" of a fixed N-term residue, and vice versa.) As such, I don't believe there should theoretically be any issues with doing partial diffusion of non-contiguous segments. 

The syntax would be analogous to the full diffusion contig syntax for loop in-painting. (e.g. `'contigmap.contigs=[9-9/A10-25/7-7/A33-56/0 B1-50]' diffuser.partial_T=20` would partially diffuse residues 1-9 and 26-32 on chain A. Note that for partial diffusion the length of the "designed" regions of the protein must match that of the input protein. (If you want to change the length and do partial diffusion, you'll need to do them in two distinct steps.)
